src/config/src/meta/destinations.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 OpenObserve Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::fmt;
+
+use hashbrown::HashMap;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub struct Destination {
+    pub id: String,
```

Comment:
I'd recommend making this an actual `svix::Ksuid` type so we can enforce at this layer that it's always valid.

If you want to make this optional so that this struct can be used as the parameter to "create" functions (where an ID would not yet have been assigned in the DB layer), then I would recommend making this `Option<svix::Ksuid>`